src/core/channelz/ztrace_collector.h

```diff
@@ -140,7 +141,22 @@ class ZTraceCollector {
       auto instance = MakeRefCounted<Instance>(std::move(args), event_engine,
                                                std::move(callback));
       auto impl = std::move(impl_);
+      RefCountedPtr<Instance> oldest_instance;
       MutexLock lock(&impl->mu);
+      if (impl->instances.size() > 20) {
```

Comment:
For my education, is this a save upperbound or do we typically expect multiple zTraces running concurrently?

Reply:
This is a very safe upperbound.

I'm thinking some automated tooling potentially, and a swarm of on-calls trying to debug something and still being way less than the limit.